geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -189,18 +189,20 @@ long zadd(Region<RedisKey, RedisData> region, RedisKey key, List<byte[]> members
     Iterator<byte[]> iterator = membersToAdd.iterator();
 
     int initialSize = getSortedSetSize();
-
+    int changesCount = 0;
     while (iterator.hasNext()) {
       byte[] score = iterator.next();
       validateScoreIsDouble(score);
       byte[] member = iterator.next();
-
       if (options.isNX() && members.containsKey(member)) {
         continue;
       }
       if (options.isXX() && !members.containsKey(member)) {
         continue;
       }
+      if (options.isCH() && members.containsKey(member)) {
+        changesCount = getChangesCount(changesCount, score, member);
+      }
```

Comment:
Since `memberAdd` returns the old value from the internal map, it might be more efficient to simply compare that to the new value and if it's different, increment `changesCount`. That way we're not doing a bunch of the same work that the map does as part of a `put()` call just to determine if the value was updated.